api.planx.uk/pay/pay.ts

```diff
@@ -0,0 +1,119 @@
+import assert from "assert";
+import { NextFunction, Request, Response } from "express";
+import { responseInterceptor } from "http-proxy-middleware";
+import SlackNotify from "slack-notify";
+import { logPaymentStatus } from "../send/helpers";
+import { usePayProxy } from "./proxy";
+import type { GovUKPayment } from "../types";
+
+assert(process.env.SLACK_WEBHOOK_URL);
+
+// exposed as /pay/:localAuthority and also used as middleware
+// returns the url to make a gov uk payment
+export async function makePaymentViaProxy(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  // confirm that this local authority (aka team) has a pay token configured before creating the proxy
+  const isSupported =
+    process.env[`GOV_UK_PAY_TOKEN_${req.params.localAuthority.toUpperCase()}`];
+
+  const flowId = req.query?.flowId as string | undefined;
+  const sessionId = req.query?.sessionId as string | undefined;
+  const teamSlug = req.params.localAuthority;
+
+  if (isSupported) {
+    // drop req.params.localAuthority from the path when redirecting
+    // so redirects to plain [GOV_UK_PAY_URL] with correct bearer token
+    usePayProxy(
+      {
+        pathRewrite: (path) => path.replace(/^\/pay.*$/, ""),
+        selfHandleResponse: true,
+        onProxyRes: responseInterceptor(
+          async (responseBuffer, _proxyRes, _req, _res) => {
+            const responseString = responseBuffer.toString("utf8");
+            const govUkResponse = JSON.parse(responseString);
+            await logPaymentStatus({
+              sessionId,
+              flowId,
+              teamSlug,
+              govUkResponse,
+            });
+            return responseBuffer;
+          }
+        ),
+      },
+      req
+    )(req, res, next);
+  } else {
+    next({
+      status: 400,
+      message: `GOV.UK Pay is not enabled for this local authority`,
+    });
+  }
+}
+
+// exposed as /pay/:localAuthority/:paymentId and also used as middleware
+// fetches the status of the payment
+export const fetchPaymentViaProxy = fetchPaymentViaProxyWithCallback(
+  async (req: Request, govUkPayment: GovUKPayment) =>
+    postPaymentNotificationToSlack(req, govUkPayment)
+);
```

Comment:
This could be written `export const fetchPaymentViaProxy = fetchPaymentViaProxyWithCallback(postPaymentNotificationToSlack)` but I think this is more explicit.